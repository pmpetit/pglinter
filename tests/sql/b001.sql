-- Test for pglinter B001 rule with file output
BEGIN;

CREATE TABLE IF NOT EXISTS my_table_without_pk (
    id INT,
    name TEXT,
    code TEXT,
    enable BOOL DEFAULT TRUE,
    query TEXT,
    warning_level INT,
    error_level INT,
    scope TEXT
);

CREATE EXTENSION IF NOT EXISTS pglinter;

-- Test with file output
SELECT pglinter.perform_base_check('/tmp/pglinter_base_results.sarif');

-- Test if file exists
\! md5sum /tmp/pglinter_base_results.sarif

-- Test with no output file (should output to prompt)
SELECT pglinter.perform_base_check();

ROLLBACK;
