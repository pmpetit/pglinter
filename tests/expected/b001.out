-- Test for pglinter B001 rule with file output
CREATE EXTENSION pglinter;
BEGIN;
CREATE TABLE IF NOT EXISTS my_table_without_pk (
    id INT,
    name TEXT,
    code TEXT,
    enable BOOL DEFAULT TRUE,
    query TEXT,
    warning_level INT,
    error_level INT,
    scope TEXT
);
-- Disable all rules first
SELECT pglinter.disable_all_rules() AS all_rules_disabled;
NOTICE:  üî¥ Disabled 21 rule(s)
 all_rules_disabled 
--------------------
                 21
(1 row)

-- Run table check to detect tables without PK
SELECT pglinter.perform_table_check();
NOTICE:  ‚úÖ No issues found - database schema looks good!
 perform_table_check 
---------------------
 t
(1 row)

-- Test rule management for B001
SELECT pglinter.explain_rule('B001');
NOTICE:  üìñ Rule Explanation for B001
============================================================

üéØ Rule Name: HowManyTableWithoutPrimaryKey
üìã Scope: BASE

üìù Description:
Count number of tables without primary key.

‚ö†Ô∏è  Message Template:
{0}/{1} table(s) without primary key exceed the {2} threshold: {3}%.

üîß How to Fix:
   1. create a primary key or change warning/error threshold
============================================================
 explain_rule 
--------------
 t
(1 row)

SELECT pglinter.is_rule_enabled('B001') AS is_b001_enabled;
 is_b001_enabled 
-----------------
 f
(1 row)

-- Test with file output
SELECT pglinter.perform_base_check('/tmp/pglinter_base_results.sarif');
 perform_base_check 
--------------------
 t
(1 row)

-- Test if file exists
\! md5sum /tmp/pglinter_base_results.sarif
7cd6073b52ee9a1173c05c016e2458f4  /tmp/pglinter_base_results.sarif
-- Test with no output file (should output to prompt)
SELECT pglinter.perform_base_check();
NOTICE:  ‚úÖ No issues found - database schema looks good!
 perform_base_check 
--------------------
 t
(1 row)

-- Re-enable B001 rule
SELECT pglinter.enable_rule('B001') AS enable_b001;
NOTICE:  ‚úÖ Rule B001 has been enabled
 enable_b001 
-------------
 t
(1 row)

-- Test again with B001 enabled
SELECT pglinter.perform_base_check();
NOTICE:  ‚úÖ No issues found - database schema looks good!
 perform_base_check 
--------------------
 t
(1 row)

-- Test with file output
SELECT pglinter.perform_base_check('/tmp/pglinter_base_results.sarif');
 perform_base_check 
--------------------
 t
(1 row)

-- Test if file exists
\! md5sum /tmp/pglinter_base_results.sarif
7cd6073b52ee9a1173c05c016e2458f4  /tmp/pglinter_base_results.sarif
ROLLBACK;
DROP EXTENSION pglinter CASCADE;
